mne/preprocessing/otp.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Authors: Samu Taulu <staulu@uw.edu>
+#          Eric Larson <larson.eric.d@gmail.com>
+
+# License: BSD (3-clause)
+
+import numpy as np
+from scipy import linalg
+
+from ..cov import _check_scalings_user, _apply_scaling_array
+from ..io.pick import pick_info, _picks_by_type, _pick_data_channels
+from ..utils import logger, verbose
+
+
+@verbose
+def oversampled_temporal_projection(raw, duration=10., scalings=None,
+                                    verbose=None):
+    """Denoise MEG channels using oversampled temporal projection
+
+    This algorithm uses a leave-one-out procedure with parallel
+    temporal projection to remove individual sensor noise under the
+    assumption that sampled fields (e.g., MEG and EEG) are oversampled
+    by the sensor array [1]_.
+
+    .. note:: Bad channels are not used or modified by this procedure.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Raw data to denoise.
```

Comment:
It's best to do it on raw data so you can use a long segment length (better performance). Doing it on epochs and evoked instances will in general underperform